File: backend/src/main/java/com/serenade/backend/domain/sync/dto/ChangesResponse.java
package com.serenade.backend.domain.sync.dto;

import com.serenade.backend.domain.playlist.dto.PlaylistSummaryResponse;
import com.serenade.backend.domain.rating.dto.RatingResponse;
import com.serenade.backend.domain.track.dto.TrackResponse;

import java.time.Instant;
import java.util.List;

public record ChangesResponse(
        List<TrackResponse> tracks,
        List<PlaylistSummaryResponse> playlists,
        List<RatingResponse> ratings,
        Instant nextCursor
) {}

File: app/app/src/main/java/com/serenade/app/feature/sync/data/remote/dto/ChangesResponse.kt
package com.serenade.app.feature.sync.data.remote.dto

import com.serenade.app.feature.playlist.data.remote.dto.PlaylistSummaryResponse
import com.serenade.app.feature.rating.data.remote.dto.RatingResponse
import com.serenade.app.feature.track.data.remote.dto.TrackResponse
import kotlinx.serialization.Serializable

@Serializable
data class ChangesResponse(
    val tracks: List<TrackResponse> = emptyList(),
    val playlists: List<PlaylistSummaryResponse> = emptyList(),
    val ratings: List<RatingResponse> = emptyList(),
    val nextCursor: String,
)
